File: src/main/java/tk/avabin/tdg/beans/entities/RPGSession.java
package tk.avabin.tdg.beans.entities;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Component;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.Set;

/**
 * Created by Avabin on 03.04.2017.
 */
@Component
@Scope("prototype")
@Data
@NoArgsConstructor
@Entity
public class RPGSession {
    @Id
    @GeneratedValue
    @NotNull
    private int id;
    private String name;
    @ManyToOne
    private User gameMaster;
    @OneToMany(targetEntity = Character.class, fetch = FetchType.EAGER)
    private Set<Character> Characters;
    @ManyToMany(targetEntity = User.class, fetch = FetchType.EAGER)
    private Set<User> players;
}

File: src/main/kotlin/tk/avabin/tdg/beans/entities/RPGSession.kt
package tk.avabin.tdg.beans.entities

import org.springframework.context.annotation.Scope
import org.springframework.stereotype.Component
import javax.persistence.Entity
import javax.persistence.FetchType
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.ManyToMany
import javax.persistence.ManyToOne
import javax.persistence.OneToMany
import javax.validation.constraints.NotNull

/**
 * Created by Avabin on 03.04.2017.
 */
@Component
@Scope("prototype")
@Entity
data class RPGSession(
        @Id
        @GeneratedValue
        @NotNull
        var id: Int = 0,
        var name: String = "",
        @ManyToOne
        var gameMaster: User? = null,
        @OneToMany(targetEntity = Character::class, fetch = FetchType.EAGER)
        var characters: Set<Character> = HashSet(),
        @ManyToMany(targetEntity = User::class, fetch = FetchType.EAGER)
        var players: Set<User> = HashSet()
)